---
title: "2020MQ044_QC"
date: "`r Sys.Date()`"
output:
    html_document:
      code_folding: hide
      toc: true
      toc_depth: 2
      highlight: tango
knit: (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(dirname(inputFile), '2020MQ044.html')) })

---


```{r setup_packages,  echo=FALSE, cache=FALSE, warning= FALSE, message=FALSE}
#method = "DDA  MBR"

install.packages('pacman')
pacman::p_load("knitr","readr","tidyverse", "stringr",
               "data.table","BiocManager", "devtools")
#I changed the way to download the packages due to an error (readRDS(dest) no connection)
install.packages("openxlsx")
#install.packages('factorextra')
install.packages('DescTools')
install.packages('magick')
devtools::install_github("hms-dbmi/UpSetR") 
BiocManager::install("SubCellBarCode")


source_file_name <- here::here("Codes",
                               "functions.R")
#testthat::test_file(here::here(
 ##   "Codes",
   # "tests",
    #"testthat",
    #"test_functions.R") )
source(source_file_name)

file_parsed = parse(source_file_name) %>% 
    Filter(is_function, .)  
unlist(Map(function_name, file_parsed))


## Global options
options(max.print="75")
opts_chunk$set(cache=FALSE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
#opts_knit$set(root.dir = "./../")
```





```{r loading_maxquant}
txt_folder <-  here::here("Datasets", "Raw",
                            "2020MQ044txt",
                            "txt")
# Quality_Control(txt_folder)
 proteus_loading <- load_MaxQuant(txt_folder)

```
We will use the qvalue package for multiple-testing correction (after looking at what the distribution of the p_values looks like) and the
https://www.bioconductor.org/packages/release/bioc/vignettes/qvalue/inst/doc/qvalue.pdf
```{r proteus analysis}
# testing <- proteus_loading$peptides$pep2prot %>% 
#   left_join(proteus_loading$peptides$stats, 
#             by = c("sequence" = "id")) %>%
#   group_split(protein, condition)
# 
# functiony <- function(x){
#   if(sum(x %>% pull(ngood))>1){
#     x
#   }else{
#     x %>% 
#       dplyr::mutate(mean = NA,
#                     variance = NA)
#   }
# }
# testing <-    purrr::map(.x = testing[1:5], functiony)
# 
# purrr::map(.x = testing[1], ~if_else(sum(.x %>% pull(ngood))>1,
#                                      .x,
#                                      rbind(.x,.x)))
PCA$
              
proteus::plotCount(proteus_loading$peptides)
peptides <- proteus_loading$peptides
proteus::plotDetectionSimilarity(peptides, bin.size = 0.02)
proteus::plotDistanceMatrix(peptides)
proteus::plotClustering(peptides)
proteus::plotPCA(peptides)
#Cpunt numbe rof peptides/sample and median
jpeg(file=here::here("Plots","pept_number.jpeg"))
proteus::plotCount(proteus_loading$peptides)
dev.off()


jpeg(file=here::here("Plots","peptJacksim.jpeg"))
proteus::plotDetectionSimilarity(peptides, bin.size = 0.02)
dev.off()                        

jpeg(file=here::here("Plots","peptPcoefmatrix.jpeg"))
proteus::plotDistanceMatrix(peptides)
dev.off()

jpeg(file=here::here("Plots","peptclustertree.jpeg"))
proteus::plotClustering(peptides)
dev.off()

jpeg(file=here::here("Plots","peptPCA.jpeg"))
proteus::plotPCA(peptides)
dev.off()
```
```{r Proteus protein analysis}
proteins <- proteus::makeProteinTable(peptides,min.peptides = 2)
summary(proteins)

#Data normalization
prodat.quant <- proteus::normalizeData(proteins, norm.fun=limma::normalizeQuantiles)
prodat1.med <- proteus::normalizeData(proteins, norm.fun=limma::normalizeMedianValues)

#Boxplot by condition
proteus::plotSampleDistributions(proteins, title="Not normalized (by condition)", fill="condition", method="box") #fill-legend colors
proteus::plotSampleDistributions(prodat1.med, title="Median normalized (by condition)", fill="condition", method="box")
proteus::plotSampleDistributions(prodat.quant, title="Quant normalized (by condition)", fill="condition", method="box")

#Box plot by sample
proteus::plotSampleDistributions(proteins, title="Not normalized (by sample)", fill="sample", method="box") #fill-legend colors
proteus::plotSampleDistributions(prodat1.med, title="Median normalized (by sample)", fill="sample", method="box")
proteus::plotSampleDistributions(prodat.quant, title="Quant normalized (by condition)", fill="sample", method="box")

proteus::plotMV(prodat1.med, with.loess=TRUE)

proteus::plotClustering(prodat1.med)



```
-There is little difference between the normalized and non normlized samples. They seem to follow a very similar distribution.
-When plotting mean against variance we see that the regresion is not linear, strange behavior.
- Changes i variance depending on the mean could affect further statistical tests (like M-A plots).

-Observation: protein-wise, sample 9 is the farthest from the others, while all other seem to cluster togheter, even going further to be clustered in their respective conditions (in exception to sample 3). However, peptide-wise, the furthest in the clustering tree was sample 8, though 9 was also in a different branch to all other samples.
```{r Protein annotation and  controls}
# protein_groups <-  proteus_loading$proteins[["stats"]]%>%
#   dplyr::select(-variance,-ngood) %>%
#  mutate(Uniprot = str_match(id,"sp\\|([:graph:]*?)\\|")[,2],
#        mean_log2 = log2(mean)) #mutate->adds new variables in the  datafrmae
# protein_groups %>% na.omit() %>% count(condition) %>%
#   ggplot(aes(x = condition, y = n)) +
#  geom_col() + ggtitle("Proteins_per_condition")


MOUSE_10090 <- read_tsv(here::here("Datasets","Raw","MOUSE_10090_idmapping.dat.gz"),col_names = FALSE) 

#MOUSE_10090_idmapping <- MOUSE_10090 %>% filter(X2 == 'Gene_Name')
# Wider_df <- protein_groups %>% ungroup() %>% distinct(condition, Uniprot, .keep_all = TRUE) %>%
#     pivot_wider(id_cols = Uniprot, names_from = "condition", values_from = "mean_log2")
# Wider_df
# 
# purrr::map2(.x = column_to_rownames( Wider_df,"Uniprot"),
#             .y = names(Wider_df %>%
#                  column_to_rownames("Uniprot")), ~hist(x  =.x ,
#                                                        main = paste("Hist of", .y)))
# length(proteus_loading$proteins$stats$id) #16000 proteins detected but only 5300 annoted (number of peptides?)
luni <- lapply(as.character(proteins$proteins), function(prot) {
 if(grepl("sp\\|", prot)) {
   uniprot <- unlist(strsplit(prot, "|", fixed=TRUE))[2]
   c(prot, uniprot)
 }#lapply applies a function to a vector, grepl finds for patterns
})
ids <- as.data.frame(do.call(rbind, luni))#makes a data table from a list
names(ids) <- c("protein", "uniprot")
annotations <- proteus::fetchFromUniProt(ids$uniprot, verbose=TRUE)
annotations.id <- merge(ids, annotations, by.x="uniprot", by.y="id")
annotations.id2 <- unique(annotations.id)
str(annotations.id)
str(annotations.id2)

prodat.med <- proteus::annotateProteins(prodat1.med, annotations.id)



```

```{r Differential expression}
#Basic differential expresion
c_sh1 <-proteus::limmaDE(prodat.med,
                         conditions=c('control','sh_1'))
c_sh2 <- proteus::limmaDE(prodat.med, 
                          conditions= c('control', 'sh_2'))
sh1_sh2 <- proteus::limmaDE(prodat.med, 
                            conditions= c('sh_1', 'sh_2'))
c_sh1

#shorter dataframe because I got stressed with the limmaDE one
c_sh1short <- c_sh1[, c('protein', 'adj.P.Val')]
c_sh1[which(c_sh1$adj.P.Val<=0.1),]

c_sh1c_sh2short <- c_sh2[, c('protein', 'adj.P.Val')]
c_sh2[which(c_sh2$adj.P.Val<=0.1),]

sh1_sh2short <- sh1_sh2[, c('protein', 'adj.P.Val')]
sh1_sh2[which(sh1_sh2$adj.P.Val<=0.1),]
sh1_sh2


#Table with adjusted P.values (BH) ascedent order
# test1 <-c_sh1short %>%
#   dplyr::arrange(desc(adj.P.Val))
# test2 <-c_sh2short[order(c_sh2short$adj.P.Val),]
# test3 <-sh1_sh2short[order(sh1_sh2short$adj.P.Val),]

#Plot the p-value distribution
proteus::plotPdist(c_sh1)
proteus::plotPdist(c_sh2)
proteus::plotPdist(sh1_sh2)

#Unique proteins in each database
#Only in control
only.cid <- which(proteins$detect$`control` & !proteins$detect$sh_1 & !proteins$detect$sh_2)
only.c <- rownames(proteins$detect[only.cid,])
only.c

#Not in control
only.sh1sh2id <- which(!proteins$detect$control & proteins$detect$`sh_1` & proteins$detect$`sh_2`)
only.sh1sh2 <-  rownames(proteins$detect[only.sh1sh2id,])
only.sh1sh2 %>% stringr::str_subset("P02340") #take a vector, detect it and only keep TRUE


#Only in p53 KD
only.sh1id <- which(!proteins$detect$control & !proteins$detect$sh_2 & proteins$detect$`sh_1`)
only.sh1 <-  rownames(proteins$detect[only.sh1id,])
length(only.sh1id)
only.sh1

#Only in RNF144b KD
only.sh2id <- which(!proteins$detect$control & !proteins$detect$sh_1 & proteins$detect$`sh_2`)
only.sh2 <-  rownames(proteins$detect[only.sh2id,])
length(only.sh2id)
only.sh2


```

```{r Control analysis}
#levels of p53/sample
summary(prodat.med)
prodat.med$tab["sp|P02340|P53_MOUSE",]
#Visulaization of peptide number and intensity per sample/condition
proteus::plotIntensities(prodat.med, id='sp|P02340|P53_MOUSE')

proteus::plotProtPeptides(peptides,'sp|P02340|P53_MOUSE', prodat.med)


#targets:: #to avoid runing everything again

#find p53 peptides accepted by proteus
seqp53<-proteus_loading$peptides$pep2prot$sequence[str_which(proteus_loading$peptides$pep2prot$protein,'P02340')]

#there are 3 detected peptides of p53 in the proteus peptide Table
#Find out in which conditions where p53 peptides found
index_pept53 <- c()
for (i in 1:3) {
    n= str_which(proteus_loading$peptides$stats$id, seqp53[i])
    print(n)
    index_pept53=c(n,index_pept53)
    print(seqp53[i])
    }
proteus_loading$peptides$stats[index_pept53,]

```

```{r}
BiocManager::install("qvalue")
library('qvalue')
qobj1 <- qvalue(p = c_sh1$P.Value, pi0=0.3) #pi0 is an estimation of true null hypothesis
summary(qobj1)
qobj2 <- qvalue(p = c_sh2$P.Value)

summary(qobj2)

subset(qobj2$qvalues, qobj2$qvalues<= 0.1)

qobj3 <- qvalue(p = sh1_sh2$P.Value)
summary(qobj3)
subset(qobj3$qvalues, qobj3$qvalues<= 0.1)
```

```{r Test of proteus peptide reading}
#Finding p53 peptides but per samples instead of conditions
seqp53<-proteus_loadingtest$peptides$pep2prot$sequence[str_which(proteus_loadingtest$peptides$pep2prot$protein,'P02340')]

#there are 3 peptides of p53
index_pept53 <- c()
for (i in 1:3) {
    n= str_which(proteus_loadingtest$peptides$stats$id, seqp53[i])
    print(n)
    index_pept53=c(n,index_pept53)
    print(seqp53[i])
    }
proteus_loadingtest$peptides$stats[index_pept53,]
```

```{r Transcriptomics}
tc_sh1 <-  read_tsv(file=here::here("Datasets", "Raw", "DESeq2ALLShrink.shRNF-shREN.txt"))
head(tc_sh1)
hist(tc_sh1$pvalue, freq = FALSE)
tc_sh2 <-  read_tsv(file=here::here("Datasets", "Raw", "DESeq2ALLShrink.shTrp53-shREN.txt"))
head(tc_sh2)
hist(tc_sh2$pvalue, freq = FALSE)


BiocManager::install('BiomaRt')

amart <- useMart('ENSEMBL_MART_MOUSE')
# select mice db mart <- useDataset('hsapiens_gene_ensembl', mart)
MOUSE_10090 <- read_tsv(here::here("Datasets","Raw","MOUSE_10090_idmapping.dat.gz"),col_names = FALSE) 
MOUSE_10090
tc_sh1
prot_genes <- MOUSE_10090 %>% dplyr::filter(X2=='Ensembl')
prot_genes <- subset(prot_genes, select=c('X1','X3'))+
testp 

n <- tc_sh1$ensambleID %>% unlist() %>% str_remove_all(pattern='\\.([:graph:]*)$')  
tc_sh1$ensambleID <- n

tc_sh1<- prot_genes %>%  left_join(y=tc_sh1,by= c('X3'='ensambleID'))
testp


ESGN1 <- tc_sh1$ensambleID
ESGN2 <- tc_sh2$EnsambleID
ESGN_annot <- getBM(
  mart = mart,   #database
  attributes = c(  #output annotation
    'ensembl_gene_id',
    'external_gene_name',
    'uniprot_gn_symbol',
    'uniprot_gn_id'),
  values= ESGN1,  #own dataset
  uniqueRows=TRUE)
#attach annotation to the initial dataframe
#create dataframe with transcriptomic+proteomic data
```
```{r Clustering}
BiocManager::install('clusterProfiler')
BiocManager::install('GO.db')
a
remove.packages(BiocManager) #needs cachem to be installed
ggo <- groupGO(gene     = gene,
               OrgDb    = org.Hs.eg.db,
               ont      = "CC",
               level    = 3,
               readable = TRUE)



```


<<<<<<< HEAD
=======
qobj3 <- qvalue(p = sh1_sh2$P.Value)
summary(qobj3)
subset(qobj3$qvalues, qobj3$qvalues<= 0.1)
<<<<<<< HEAD
```
>>>>>>> f754664468f840c56716727458d8563ad61aff8a
=======
```
>>>>>>> f754664468f840c56716727458d8563ad61aff8a
